File: Rentify Server/models/Support.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const Support = new Schema({ 
    user_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    room_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Room',
    },
    landlord_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
    },
    title_support: { type: String, require: true },
    content_support: { type: String, require: true },
    image: { type: Array },
    status: { type: Number, enum: [0, 1], default: 1 },
    created_at: { type: String },
    updated_at: { type: String }
})
module.exports = mongoose.model("Support", Support);

File: Rentify/app/src/main/java/com/rentify/user/app/model/SupportModel/Support.kt
package com.rentify.user.app.model.SupportModel

import com.google.gson.annotations.SerializedName

data class SupportResponse(
    @SerializedName("_id")val _id: String,
    val building_id: BuildingOfSupport,
    val content_support: String,
    val created_at: String,
    val image: List<String>,
    val room_id: RoomOfSupport,
    val status: Int,
    val title_support: String,
    val updated_at: String,
    val user_id: UserOfSupport
)

data class RoomOfSupport(
    @SerializedName("_id")val _id: String,
    val room_name: String
)

data class UserOfSupport(
    @SerializedName("_id")val _id: String
)

data class BuildingOfSupport(
    @SerializedName("_id")val _id: String,
    val nameBuilding: String
)

data class CreateReportResponse(
    val message: String,
    val support: Any
)